pkg/dbx/update.go

```diff
@@ -0,0 +1,185 @@
+	for i, setClause := range d.SetClauses {
+		var valSql string
+		if vs, ok := setClause.value.(Sqlizer); ok {
+			vsql, vargs, err := vs.ToSql()
+			if err != nil {
+				return "", nil, err
+			}
+			valSql = vsql
+			args = append(args, vargs...)
+		} else {
+			valSql = "?"
+			args = append(args, setClause.value)
+		}
+		setSqls[i] = fmt.Sprintf("%s = %s", setClause.column, valSql)
+	}
+	sql.WriteString(strings.Join(setSqls, ", "))
+
+	if len(d.WhereParts) > 0 {
+		sql.WriteString(" WHERE ")
+		args, err = appendToSql(d.WhereParts, sql, " AND ", args)
+		if err != nil {
+			return
+		}
+	}
+
+	if d.Limit != "" {
+		sql.WriteString(" LIMIT ")
+		sql.WriteString(d.Limit)
+	}
+
+	sqlStr, err = d.PlaceholderFormat.ReplacePlaceholders(sql.String())
+
+	return
+}
+
+// Builder
+
+// UpdateBuilder[T] builds SQL UPDATE statements.
+type UpdateBuilder[T any] builder.Builder
+
+func newUpdateBuilder[T any](client db.Client, table string, placeholderFormat PlaceholderFormat) UpdateBuilder[T] {
+	b := builder.Builder(builder.EmptyBuilder)
+	ub := UpdateBuilder[T](b).table(table)
+	ub = ub.placeholderFormat(placeholderFormat)
+	ub = builder.Set(ub, "Client", client).(UpdateBuilder[T])
+
+	return ub
+}
+
+func (b UpdateBuilder[T]) Exec(ctx context.Context) (sql.Result, error) {
+	data := builder.GetStruct(b).(updateData[T])
+
+	return data.Exec(ctx)
+}
+
+// Format methods
+
+// PlaceholderFormat sets PlaceholderFormat (e.g. Question or Dollar) for the
+// query.
+func (b UpdateBuilder[T]) placeholderFormat(f PlaceholderFormat) UpdateBuilder[T] {
+	return builder.Set(b, "PlaceholderFormat", f).(UpdateBuilder[T])
+}
+
+// SQL methods
+
+// Table sets the table to be updated.
+func (b UpdateBuilder[T]) table(table string) UpdateBuilder[T] {
+	return builder.Set(b, "Table", table).(UpdateBuilder[T])
+}
+
+// Set adds SET clauses to the query.
+func (b UpdateBuilder[T]) Set(column string, value any) UpdateBuilder[T] {
+	return builder.Append(b, "SetClauses", setClause{column: column, value: value}).(UpdateBuilder[T])
+}
+
+// Set adds SET clauses to the query.
+func (b UpdateBuilder[T]) Options(options ...string) UpdateBuilder[T] {
+	return builder.Extend(b, "Options", options).(UpdateBuilder[T])
+}
+
+// SetMap is a convenience method which calls .Set for each key/value pair in clauses.
+func (b UpdateBuilder[T]) SetMap(clauses map[string]any) UpdateBuilder[T] {
+	keys := make([]string, len(clauses))
+	i := 0
+	for key := range clauses {
+		keys[i] = key
+		i++
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		val := clauses[key]
+		b = b.Set(key, val)
+	}
+
+	return b
+}
+
+// Where adds WHERE expressions to the query.
+//
+// See SelectBuilder.Where for more information.
+func (b UpdateBuilder[T]) Where(pred any, args ...any) UpdateBuilder[T] {
+	return applyWhere[T](b, pred, args...).(UpdateBuilder[T])
+}
+
+// Limit sets a LIMIT clause on the query.
+func (b UpdateBuilder[T]) Limit(limit uint64) UpdateBuilder[T] {
```

Comment:
I just noticed that the capabilities may depend on the dialect / version. On 8.4, Mysql supports also order by for the update statement

Reply:
brought it back..